### PR TITLE
Feat: Add Logrocket Monitoring

### PR DESCRIPTION
**Pull request by Cliftonz**
### What change does this PR introduce?
Add LogRocket monitoring to get better analytics of what our users are using in the front-end.

### CODE REVIEW

File: apps/web/src/App.tsx [hunk 1]
```diff
@@ -36,6 +36,13 @@ import { LinkVercelProjectPage } from './pages/partner-integrations/LinkVercelPr
 import { useBlueprint } from './hooks/useBlueprint';
 import { BrandPage } from './pages/brand/BrandPage';
 import { SegmentProvider } from './store/segment.context';
+import LogRocket from 'logrocket';
+import setupLogRocketReact from 'logrocket-react';
+
+if (LOGROCKET_ID) {
```
Comment (LetItRock) [inline apps/web/src/App.tsx:42]: do we want to have it enabled during development?

File: apps/web/public/env-config.js [hunk 2]
```diff
@@ -1,3 +1,4 @@
 window._env_ = {
   SKIP_PREFLIGHT_CHECK: 'true',
+  REACT_APP_ENVIRONMENT: 'dev',
```
Comment (LetItRock) [inline apps/web/public/env-config.js:3]: as far as I know, this file is generated from the `.env` and we do override `REACT_APP_ENVIRONMENT` in the deploy pipeline... as far as I see it's already set in the `.env`

File: apps/web/.env [hunk 3]
```diff
@@ -6,3 +6,4 @@ GLOBAL_CONTEXT_PATH=
 WEB_CONTEXT_PATH=
 REACT_APP_INTERCOM_APP_ID=
 REACT_APP_SEGMENT_KEY=
+LOGROCKET_ID='eh8t7r/novu'
```
Comment (LetItRock) [inline apps/web/.env:9]: shouldn't we set it from the pipeline?

Comment (scopsy) [reply apps/web/.env:9]: Yes, this should be removed from the .env file. And added to the deploy dev web and deploy prod web workflows on github actions.

File: apps/web/public/env-config.js (hunk 2 above)
Comment (scopsy) [reply apps/web/public/env-config.js:3]: Yes, totally correct. This file shouldn't be modified.

File: apps/web/src/App.tsx (hunk 1 above)
Comment (scopsy) [reply apps/web/src/App.tsx:42]: I don't think so, @LetItRock do you see any benefits of it?

File: package.json [hunk 4]
```diff
@@ -48,7 +48,6 @@
     "build:web": "nx build @novu/web",
     "build:widget": "nx build @novu/widget",
     "build:embed": "nx build @novu/embed",
-    "build:inbound-mail": "nx build @novu/inbound-mail",
```
Comment (scopsy) [inline package.json:51]: Why was this removed from here?

Comment (scopsy) [reply package.json:51]: Oh nevermind, just seen this was a duplicate 😅

File: apps/web/src/App.tsx (hunk 1 above)
Comment (Cliftonz) [reply apps/web/src/App.tsx:42]: I don't think it has any benefit in dev, the analytics we need from this are only in prod.

**Review by scopsy — APPROVED** [review]
